rules/Php81/NodeFactory/EnumFactory.php

```diff
@@ -91,14 +96,76 @@ private function createEnumCaseFromConst(ClassConst $classConst): EnumCase
         return $enumCase;
     }
 
-    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode): EnumCase
+    /**
+     * @param array<string, int|string> $mapping
+     */
+    private function createEnumCaseFromDocComment(PhpDocTagNode $phpDocTagNode, array $mapping = []): EnumCase
     {
         /** @var MethodTagValueNode $nodeValue */
         $nodeValue = $phpDocTagNode->value;
-
+        $enumValue = $mapping[$nodeValue->methodName] ?? $nodeValue->methodName;
         $enumName = strtoupper($nodeValue->methodName);
-        $enumExpr = $this->builderFactory->val($nodeValue->methodName);
+        $enumExpr = $this->builderFactory->val($enumValue);
 
         return new EnumCase($enumName, $enumExpr);
     }
+
+    /**
+     * @return array<string, int|string>
+     */
+    private function generateMappingFromClass(Class_ $class): array
+    {
+        $mapping = [];
+
+        $classMethod = $this->getValuesClassMethod($class);
+        if ($classMethod === null || ! is_array($classMethod->stmts)) {
+            return $mapping;
+        }
+
+        foreach ($classMethod->stmts as $methodStmt) {
+            if (
+                ! ($methodStmt instanceof Return_)
+                || ! ($methodStmt->expr instanceof Expr\Array_)
+            ) {
+                continue;
+            }
+            foreach ($methodStmt->expr->items as $item) {
+                if ($item instanceof Expr\ArrayItem) {
+                    $mapping[$item->key->value] = $item->value->value;
+                }
+            }
+        }
+
+        return $mapping;
+    }
+
+    private function getValuesClassMethod(Class_ $class): ?ClassMethod
+    {
+        foreach ($class->stmts as $stmt) {
+            if ($stmt instanceof ClassMethod && $stmt->name->name === 'values') {
+                return $stmt;
+            }
+        }
+
+        return null;
```

Comment:
You can directly `return $class->getMethod('values')`

```suggestion
        return $class->getMethod('values')
```